spring-cloud-gcp-data-datastore/src/main/java/org/springframework/cloud/gcp/data/datastore/core/DatastoreTemplate.java

```diff
@@ -184,12 +186,28 @@ public long count(Class<?> entityClass) {
 
 	@Override
 	public <T> Collection<T> findAll(Class<T> entityClass) {
-		return convertEntitiesForRead(
-				this.datastore.run(Query.newEntityQueryBuilder()
-						.setKind(this.datastoreMappingContext
-								.getPersistentEntity(entityClass).kindName())
-						.build()),
-				entityClass);
+		return findAll(entityClass, -1, -1, null);
+	}
+
+	@Override
+	public <T> Collection<T> findAll(Class<T> entityClass, int limit, int offset,
+			Sort sort) {
+		DatastorePersistentEntity<?> persistentEntity = this.datastoreMappingContext.getPersistentEntity(entityClass);
+		EntityQuery.Builder builder = Query.newEntityQueryBuilder()
+				.setKind(persistentEntity.kindName());
+		if (limit > 0) {
```

Comment:
Is there a way to use the code from `PartTreeDatastoreQuery` or to merge the code here in? That one already supports limiting and sorting, and I don't think we should have logic for building a query with sort and limits in 2 places.